app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -597,19 +599,36 @@ public boolean onLongClick(final View v) {
     }
 
     private void toggleTitleAndSecondaryControls() {
-        if (binding.detailSecondaryControlPanel.getVisibility() == View.GONE) {
-            binding.detailVideoTitleView.setMaxLines(10);
-            animateRotation(binding.detailToggleSecondaryControlsView,
-                    VideoPlayerUi.DEFAULT_CONTROLS_DURATION, 180);
-            binding.detailSecondaryControlPanel.setVisibility(View.VISIBLE);
+        if (binding.detailVideoTitleView instanceof NewPipeTextView) {
+            if (binding.detailSecondaryControlPanel.getVisibility() == View.GONE) {
+                AnimationUtil.expand(binding.detailVideoTitleView, 500, 10);
+                animateRotation(binding.detailToggleSecondaryControlsView,
+                        VideoPlayerUi.DEFAULT_CONTROLS_DURATION, 180);
+                AnimationUtil.expand(binding.detailSecondaryControlPanel, 500,
+                        (animatedFraction, isCollapsing) -> updateTabLayoutVisibility());
+            } else {
+                AnimationUtil.collapse(binding.detailVideoTitleView, 500, 1);
+                animateRotation(binding.detailToggleSecondaryControlsView,
+                        VideoPlayerUi.DEFAULT_CONTROLS_DURATION, 0);
+                AnimationUtil.collapse(binding.detailSecondaryControlPanel, 500,
+                        (animatedFraction, isCollapsing) -> updateTabLayoutVisibility());
+            }
         } else {
-            binding.detailVideoTitleView.setMaxLines(1);
-            animateRotation(binding.detailToggleSecondaryControlsView,
-                    VideoPlayerUi.DEFAULT_CONTROLS_DURATION, 0);
-            binding.detailSecondaryControlPanel.setVisibility(View.GONE);
+            // legacy approach
```

Comment:
I think you can safely assume that `binding.detailVideoTitleView` will always be instance of `NewPipeTextView`. Also, I wonder why the binding element is not already cast to that type. Maybe check if the text view has type `NewPipeTextView` both in the normal and the large-land layouts